src/Collie/Modifiers.idr

```diff
@@ -12,12 +12,12 @@ public export
 
```

Comment:
Can we not keep `String` everywhere here and implement a `reflow` function
that takes a `String`, `words` it and then cuts it up in ~80 characters lines?

Something like this:

```idris
import Data.SnocList
import Data.String

reflow : (width : Nat) -> String -> List String
reflow w = map concat . init . words

  where

    init : List String -> List (List String)
    go : Nat -> SnocList String -> List String -> List (List String)

    init [] = []
    init (x :: xs) = go (minus w (length x)) [< x] xs

    go Z acc xs        = (acc <>> []) :: init xs
    go n acc []        = [acc <>> []]
    go n acc (x :: xs) = go (minus n (length x)) (acc :< x) xs
```

Or even

```idris
    go n acc (x :: xs) =
      let l = length x in
      if n < l then (acc <>> []) :: init (x :: xs)
      else go (minus n l) (acc :< x) xs
```
if we want to guarantee we stay under the width (as long as no single word is larger
than the given width of course...)

Reply:
I'm not sure what the best option is. I expect sometimes users would want to keep their own line breaks (for example, displaying a URL or some ASCII art ) and sometimes they'd rather reflow.

I figured using a list of strings is probably the best since users could use a function to `reflow` a string into a list of lines if they wanted to. The only information `usage` has that they won't have is the overall nesting, which affects the computation of the maximum width to reflow to.

Reply:
OK, I reverted back to String, and we now reflow lines that are longer than the newly added `maxWidth` parameter (defaulting to 80).

Reply:
Ah I guess my code was broken but why not have `reflow w = map unwords . init . words`
instead of interspersing spaces only to remove the leading ones?

Reply:
Ah, I guess you have to use `minus n (S l)` in the `go` function too!

Reply:
Ah, forgot about `unwords`.

Reply:
Done now!